OrrRock/OrrRock/RouteFindingUpload/RouteFindingCameraViewController.swift

```diff
@@ -0,0 +1,300 @@
+        return button
+    }()
+    
+    private lazy var closeButton: UIButton = {
+        let button = UIButton()
+        button.backgroundColor = UIColor(hex: "000000").withAlphaComponent(0.3)
+        button.layer.borderColor = UIColor(hex: "979797").cgColor
+        button.layer.borderWidth = 1
+        button.layer.cornerRadius = 20
+        let config = UIImage.SymbolConfiguration(pointSize: 24, weight: .medium, scale: .small)
+        let buttonSymbol = UIImage(systemName: "multiply", withConfiguration: config)?.withTintColor(UIColor.orrWhite ?? UIColor.white, renderingMode: .alwaysOriginal)
+        button.setImage(buttonSymbol, for: .normal)
+        return button
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setUpLayout()
+        setPhotosButtonImage()
+        
+        setCameraPreviewLayer()
+        authorizateCameraStatus()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        photoImage = nil
+        photoData = nil
+        currentLocalIdentifier = nil
+        
+        executeCameraSession()
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(setPhotosButtonImage), name: UIApplication.willEnterForegroundNotification, object: nil)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        
+        if cameraAuthorizeStatus == .success {
+            captureSession.stopRunning()
+        }
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        
+        NotificationCenter.default.removeObserver(self, name: UIApplication.willEnterForegroundNotification, object: nil)
+    }
+}
+
+// MARK: Layout Function
+private extension RouteFindingCameraViewController {
+    func setUpLayout() {
+        
+        let shutterButtonSize: CGFloat = 75
+        let safeArea = view.safeAreaLayoutGuide
+        let height: CGFloat = UIScreen.main.bounds.width * 16/9
+        view.addSubview(cameraView)
+        cameraView.snp.makeConstraints({
+            $0.top.equalTo(view.snp.top)
+            $0.leading.equalTo(view.snp.leading)
+            $0.trailing.equalTo(view.snp.trailing)
+            $0.height.equalTo(height)
+        })
+        
+        view.addSubview(shutterButton)
+        shutterButton.snp.makeConstraints({
+            $0.bottom.equalTo(safeArea.snp.bottom).inset(OrrPd.pd16.rawValue)
+            $0.centerX.equalTo(safeArea.snp.centerX)
+            $0.width.height.equalTo(shutterButtonSize)
+        })
+        
+        let circleLayer = CAShapeLayer()
+        let circleSize: CGFloat = 57
+        let circleShape = UIBezierPath(ovalIn: CGRect(x: (shutterButtonSize - circleSize)/2, y: (shutterButtonSize - circleSize)/2, width: circleSize, height: circleSize))
```

Comment:
```suggestion
        let circleShapeSize = (shutterButtonSize - circleSize)/2
        let circleShape = UIBezierPath(ovalIn: CGRect(x: circleShapeSize, y: circleShapeSize, width: circleSize, height: circleSize))
```

똑같은 연산을 2번 하는건 별로일듯해요

Reply:
리뷰 감사합니다 코인! 44fb652808038cdc4b889e7b84740393c5858080 에 반영했습니다 :)